core/genesis.go

```diff
@@ -310,6 +310,12 @@ func (g *Genesis) ToBlock(db ethdb.Database) *types.Block {
 			head.BaseFee = g.Config.GetFeeConfig().MinBaseFee
 		}
 	}
+	if g.Config.FeeConfig.GasLimit.Uint64() != head.GasLimit {
+		panic(fmt.Sprintf(
+			"gas limit in fee config (%d) does not match gas limit in header (%d)",
+			g.Config.FeeConfig.GasLimit, head.GasLimit,
+		))
+	}
```

Comment:
Can we handle this within `SetupGenesisBlock` instead of `ToBlock` so that we can return a fatal error instead of panicking here?